### PR TITLE
Bridge and cross chain messaging docs.

### PR DESCRIPTION
**Pull request by StefanIliev545**
### Why this change is needed

Please provide a description and a link to the underlying ticket

### What changes were made as part of this PR

Please provide a high level list of the changes made

### PR checks pre-merging

Please indicate below by ticking the checkbox that you have read and performed the required
[PR checks](https://github.com/obscuronet/obscuro-internal/blob/main/dev-ops-docs/dev-pr-checks.md)

- [ ] PR checks reviewed and performed

### CODE REVIEW

File: docs/_docs/cross-chain/bridge.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+The value received by the bridge contract during the execution of `sendNative` is what will be logged as transfer to the other layer.
+
+## Layer 1 To Layer 2 Specifics
+
+The `ObscuroBridge.sol` contract is responsible for managing the layer 1 side of the bridge. It's the "bridge to Obscuro".
+The `EthereumBridge.sol` contract is responsible for managing the layer 2 side of the bridge. It's the "bridge to Ethereum".
+
+In order to bridge tokens over they need to be whitelisted. **Only accounts with the admin role can whitelist tokens!**
+
+When an asset is whitelisted, the bridge internally uses the `publishMessage` call on the MessageBus which can be found within the `ManagementContract`. In the message that is published the bridge "tells" the other side of it, which resides on layer 2 that a token has been whitelisted. This in turn creates a wrapped version of the token on the other side. This version of the token can only be minted and burned by the layer 2 bridge.
```
Comment (moraygrieve) [inline docs/_docs/cross-chain/bridge.md:29]: I don't think publishMessage is in the management contract is it?

Comment (moraygrieve) [inline docs/_docs/cross-chain/bridge.md:29]: Do we also need to mention that a relay needs to be in place here?

File: docs/_docs/cross-chain/bridge.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+Interacting with `sendNative` does not require approving an allowance, but instead just add value to the transaction (or evm call).
+The value received by the bridge contract during the execution of `sendNative` is what will be logged as transfer to the other layer.
+
+## Layer 1 To Layer 2 Specifics
+
+The `ObscuroBridge.sol` contract is responsible for managing the layer 1 side of the bridge. It's the "bridge to Obscuro".
+The `EthereumBridge.sol` contract is responsible for managing the layer 2 side of the bridge. It's the "bridge to Ethereum".
+
+In order to bridge tokens over they need to be whitelisted. **Only accounts with the admin role can whitelist tokens!**
```
Comment (moraygrieve) [inline docs/_docs/cross-chain/bridge.md:27]: Do we / or can we mention here the process of requesting an ERC20 to be whitelisted if only accounts with the admin role can do this i.e. if I am a user of Obscuro and want to bridge my token, who do I ask to perform the whitelist?

File: docs/_docs/cross-chain/bridge.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+When an asset is whitelisted, the bridge internally uses the `publishMessage` call on the MessageBus which can be found within the `ManagementContract`. In the message that is published the bridge "tells" the other side of it, which resides on layer 2 that a token has been whitelisted. This in turn creates a wrapped version of the token on the other side. This version of the token can only be minted and burned by the layer 2 bridge.
+
+ * Minting allows to create fresh funds on the L2 when they get locked on L1.
+ * Burning allows to destroy supply on L2 in order to release it from the bridge on L1.
+
+The protocol to bridge assets using `mint`/`burn` is based on the `MessageBus`'s `publishMessage` too. The `ObscuroBridge` tells the layer 2 that it has locked a certain amount of tokens and because of that address Y should receive the same amount of the wrapped token counterpart. This is when `mint` comes into play. On the other hand, the `EthereumBridge` burns tokens when received and tells the L1 that it has in fact done so. This is interpreted by the L1 smart contract as evidence to release locked funds.
+
```
Comment (moraygrieve) [inline docs/_docs/cross-chain/bridge.md:35]: the published message still needs to be relayed again etc

File: docs/_docs/cross-chain/bridge.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+## Security
+
+The bridge inherits the `CrossChainEnabled` contract. It provides a modifier that can be attached to public smart contract functions in order to limit them. Normally you can limit a function to be called by `msg.sender`. With this new modifier you can limit a function to be called from the equivalent of `msg.crosschainsender`.
```
Comment (moraygrieve) [inline docs/_docs/cross-chain/bridge.md:39]: Should this not be the CrossChainEnabledObscuro contract?

File: docs/_docs/cross-chain/bridge.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+## Building Bridges
+
+The standard bridge was designed to work without requiring access to any private platform capabilities. This means that anyone can build a bridge using the same cross chain messaging API. There is no permissioning required to do so.
+
+## Interface 
+
+```solidity
+interface IBridge {
+    function sendNative(address receiver) external payable;
```
Comment (moraygrieve) [inline docs/_docs/cross-chain/bridge.md:54]: I was going to ask, but why no amount here?

File: docs/_docs/cross-chain/core-messaging-layer.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+---
+# Obscuro Cross Chain Messaging
+
+Obscuro is equipped with a cross chain messaging protocol that enables sending and receiving messages across layers securely and in an entirely decentralized fashion.
+
+The core contract that provides this functionality is the `MessageBus`. It exists both on L1 and L2. In the L1 you can find it under the management contract whilst on the L2 it is created and managed by the enclave.
+
+## How It Works
+
+Users call the `publishMessage` function on the `MessageBus`. This function emits an event that records what message is being published, who is calling the method and some other bits. This event later on gets picked on by the enclave which ensures that the message is available on the counterpart `MessageBus`.
```
Comment (moraygrieve) [inline docs/_docs/cross-chain/core-messaging-layer.md:11]: Is it worth detailing how it gets picked up i.e. (if my understanding is correct) the enclave consume the blocks from the L1 and reads the message as an event within the transaction logs?

File: docs/_docs/cross-chain/core-messaging-layer.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+When the message is made available, users can call the `verifyMessageFinalized` method in the `MessageBus` on the receipient layer. This method will return `true` if the message was indeed sent on the other layer as presented. Same `msg.sender`, same `payload`, same `nonce`.
+
+This allows to have a behaviour much like querying questions - **'Has the bridge on address 0xAB0FF received 25 WETH tokens?'**. If the bridge on this address has called `publishMessage` saying 'I have received 25 WETH tokens?` the query will return true.
```
Comment (moraygrieve) [inline docs/_docs/cross-chain/core-messaging-layer.md:15]: Quotes are wrong here 

'I have received 25 WETH tokens?`

File: docs/_docs/cross-chain/core-messaging-layer.md [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+
+Obscuro is equipped with a cross chain messaging protocol that enables sending and receiving messages across layers securely and in an entirely decentralized fashion.
+
+The core contract that provides this functionality is the `MessageBus`. It exists both on L1 and L2. In the L1 you can find it under the management contract whilst on the L2 it is created and managed by the enclave.
+
+## How It Works
+
+Users call the `publishMessage` function on the `MessageBus`. This function emits an event that records what message is being published, who is calling the method and some other bits. This event later on gets picked on by the enclave which ensures that the message is available on the counterpart `MessageBus`.
+
+When the message is made available, users can call the `verifyMessageFinalized` method in the `MessageBus` on the receipient layer. This method will return `true` if the message was indeed sent on the other layer as presented. Same `msg.sender`, same `payload`, same `nonce`.
```
Comment (moraygrieve) [inline docs/_docs/cross-chain/core-messaging-layer.md:13]: Is it worth saying you effectively need to construct the cross chain message from the transaction logs in order to call verifyMessageFinalized? Basically are we trying to keep this doc fairly high level or give more granular detail of the various call structures.

File: docs/_docs/cross-chain/bridge.md (hunk 4 above)
Comment (tudor-malene) [inline docs/_docs/cross-chain/bridge.md:39]: what modifier? Also an example would be good

File: docs/_docs/cross-chain/bridge.md [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+
+
+## Security
+
+The bridge inherits the `CrossChainEnabled` contract. It provides a modifier that can be attached to public smart contract functions in order to limit them. Normally you can limit a function to be called by `msg.sender`. With this new modifier you can limit a function to be called from the equivalent of `msg.crosschainsender`. 
+
+The bridge links its two sides together - the ObscuroBridge and the EthereumBridge and limits the cross chain sender to only be the opposite contract on the `receiveAssets` functions. This means that only the `CrossChainMessenger` can call them when it is provided with a valid cross chain message produced by the bridge. As the cross chain sender is determined by `msg.sender` at the point of calling `publishMessage`, no one else but the bridge has the capability of queueing a message that will pass.
```
Comment (tudor-malene) [inline docs/_docs/cross-chain/bridge.md:41]: hard to follow para

File: docs/_docs/cross-chain/bridge.md [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+
+The layer 2 bridge works much the same way as the L1 bridge and uses the `MessageBus` to facilitate communication. 
+
+## Building Bridges
+
+The standard bridge was designed to work without requiring access to any private platform capabilities. This means that anyone can build a bridge using the same cross chain messaging API. There is no permissioning required to do so.
+
+## Interface 
+
+```solidity
```
Comment (tudor-malene) [inline docs/_docs/cross-chain/bridge.md:52]: a bit more context

File: docs/_docs/cross-chain/bridge.md [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+
+The bridge links its two sides together - the ObscuroBridge and the EthereumBridge and limits the cross chain sender to only be the opposite contract on the `receiveAssets` functions. This means that only the `CrossChainMessenger` can call them when it is provided with a valid cross chain message produced by the bridge. As the cross chain sender is determined by `msg.sender` at the point of calling `publishMessage`, no one else but the bridge has the capability of queueing a message that will pass.
+
+
+The layer 2 bridge works much the same way as the L1 bridge and uses the `MessageBus` to facilitate communication. 
+
+## Building Bridges
+
+The standard bridge was designed to work without requiring access to any private platform capabilities. This means that anyone can build a bridge using the same cross chain messaging API. There is no permissioning required to do so.
```
Comment (tudor-malene) [inline docs/_docs/cross-chain/bridge.md:48]: mention that we encourage developers to build novel bridges based on liquidity pools or novel concepts.
And that we provide all the apis they need

**Review by tudor-malene — CHANGES_REQUESTED** [review]
thanks Stefan. Left a few comments

File: docs/_docs/cross-chain/core-messaging-layer.md (hunk 6 above)
Comment (StefanIliev545) [reply docs/_docs/cross-chain/core-messaging-layer.md]: I've tried to avoid in depth detail as I think it might be better suited inside the whitepaper/tech design (or another docs section really).

File: docs/_docs/cross-chain/core-messaging-layer.md (hunk 8 above)
Comment (StefanIliev545) [reply docs/_docs/cross-chain/core-messaging-layer.md:13]: Yep, trying for this to be fairly high level. My expectation is that an automated relayer will pop up at some point so the low level detail will be hidden away.

File: docs/_docs/cross-chain/core-messaging-layer.md [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+---
+# Obscuro Cross Chain Messaging
+
+Obscuro is equipped with a cross chain messaging protocol that enables sending and receiving messages across layers securely and in an entirely decentralized fashion.
+
+The core contract that provides this functionality is the `MessageBus`. It exists both on L1 and L2. In the L1 you can find it under the management contract whilst on the L2 it is created and managed by the protocol.
+
+## How It Works
+
+Users call the `publishMessage` function on the `MessageBus`. This function emits an event that records what message is being published, who is calling the method and some other bits. This event later on gets picked on by the protocol which ensures that the message is available on the counterpart `MessageBus`.
```
Comment (moraygrieve) [inline docs/_docs/cross-chain/core-messaging-layer.md:11]: 'picked on' or should that read 'picked up by'?

File: docs/_docs/cross-chain/core-messaging-layer.md [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+
+When the message is made available, users can call the `verifyMessageFinalized` method in the `MessageBus` on the receipient layer. This method will return `true` if the message was indeed sent on the other layer as presented. Same `msg.sender`, same `payload`, same `nonce`.
```
Comment (moraygrieve) [inline docs/_docs/cross-chain/core-messaging-layer.md:13]: receipient should read recipient

**Review by moraygrieve — APPROVED** [review]
LGTM
